geode-wan/src/test/java/org/apache/geode/cache/client/internal/locator/wan/LocatorMembershipListenerTest.java

```diff
@@ -92,9 +94,19 @@ private void verifyMessagesSentBothWays(DistributionLocatorId sourceLocator,
 
   private void joinLocatorsDistributorThread() throws InterruptedException {
     Set<Thread> threads = Thread.getAllStackTraces().keySet();
+
+    // Wait for the daemon thread to be alive.
+    try {
+      await().untilAsserted(() -> assertThat(
```

Comment:
In try block, the "threads" is not getting re-set; it will be looking at the same set of entries. You may want to move getting threads in the wait loop.

Reply:
Great catch @agingade, thanks!.